### PR TITLE
MMT-3496: Making PlatformField functional

### PR DESCRIPTION
**Pull request by mandyparson**
Under Collection Drafts >> Acquisition Information >> Platforms >> Short Name
You can select any option and it will auto fill the other fields. Clicking "Clear Short Name" will clear the options.
Screenshot 2024-03-13 at 3 00 42 PM

### CODE REVIEW

File: static/src/js/components/PlatformField/PlatformField.jsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+import './PlatformField.scss'
+
+/** This is custom field found with Collections >> acquisitionInformation
+ *
+ * @typedef {Object} PlatformField
+ * @property {Function} onChange A callback function triggered when the user inputs a text.
+ * @property {Object} uiSchema uiSchema for the field being shown.
+ * @property {Object} formData Saved Draft
+ */
+const PlatformField = (props) => {
```
Comment (dmistry1) [inline static/src/js/components/PlatformField/PlatformField.jsx:15]: Pull out `onChange`, `uiSchema` and `formData` from the the props and only pass those values into the component.

Look at PreviewMapTemplate as an example

File: static/src/js/components/PlatformField/PlatformField.jsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+  const { formData } = props
+  const { Type = '', ShortName = '', LongName = '' } = formData
+  const [longNameMap, setLongNameMap] = useState({})
+  const [state, setState] = useState({
```
Comment (dmistry1) [inline static/src/js/components/PlatformField/PlatformField.jsx:19]: Pull all of these state variables into their own const state variables

File: static/src/js/components/PlatformField/PlatformField.jsx [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+          onBlur={onHandleBlur}
+          menuIsOpen={state.showMenu}
+        />
+      </div>
+
+      {/* Type field */}
+      <div style={{ marginTop: '8px' }}>
+        Type
+        <input
+          className="platForm-field-text-field"
```
Comment (dmistry1) [inline static/src/js/components/PlatformField/PlatformField.jsx:217]: Follow the new scss className naming convention, look at CustomWidgetWrapper or ProgressSection as an example.

File: static/src/js/components/PlatformField/PlatformField.jsx [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+/** This is custom field found with Collections >> acquisitionInformation
+ *
+ * @typedef {Object} PlatformField
+ * @property {Function} onChange A callback function triggered when the user inputs a text.
+ * @property {Object} uiSchema uiSchema for the field being shown.
+ * @property {Object} formData Saved Draft
+ */
+const PlatformField = ({ onChange, uiSchema, formData }) => {
+  const { Type = '', ShortName = '', LongName = '' } = formData
+  const [type] = useState(Type)
```
Comment (dmistry1) [inline static/src/js/components/PlatformField/PlatformField.jsx:17]: Not quite, something like this
`const [type, setType] = useState()`

File: static/src/js/components/PlatformField/PlatformField.jsx [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+      setLoading(false)
+      setKeyword(newPaths)
+    }
+
+    fetchData()
+  }, [])
+
+  const onHandleMouseDown = (values) => {
+    const [valueType, valueShortName] = values
+    const valueLongName = longNameMap[valueShortName] || 'No Long Name Provided'
```
Comment (tle1989) [inline static/src/js/components/PlatformField/PlatformField.jsx:74]: I found a discrepancy here regarding to including the Long Name wording is different, and is included in JSON compared to umm_tool_prototype; other than that it looks good.
Screenshot 2024-03-20 at 2 51 10 PM

Screenshot 2024-03-20 at 2 50 50 PM

**Review by tle1989 — APPROVED** [review]

**Pull request merged**